tests/components/shelly/test_init.py

```diff
@@ -392,6 +395,49 @@ async def test_entry_missing_gen(hass: HomeAssistant, mock_block_device: Mock) -
     assert hass.states.get("switch.test_name_channel_1").state is STATE_ON
 
 
+async def test_entry_missing_port(hass: HomeAssistant) -> None:
+    """Test successful Gen2 device init when port is missing in entry data."""
+    with patch(
+        "aioshelly.rpc_device.RpcDevice.create", return_value=Mock()
+    ) as rpc_device_mock:
+        data = {
+            CONF_HOST: "192.168.1.37",
+            CONF_SLEEP_PERIOD: 0,
+            "model": MODEL_PLUS_2PM,
+            CONF_GEN: 2,
+        }
+        entry = MockConfigEntry(domain=DOMAIN, data=data, unique_id=MOCK_MAC)
+        entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert rpc_device_mock.call_args[0][2] == ConnectionOptions(
+            ip_address="192.168.1.37", device_mac="123456789ABC", port=80
+        )
+
+
+async def test_rpc_entry_custom_port(hass: HomeAssistant) -> None:
+    """Test successful Gen2 device init using custom port."""
+    with patch(
+        "aioshelly.rpc_device.RpcDevice.create", return_value=Mock()
```

Comment:
Usually we try to patch where RpcDevice is imported

Reply:
Done